### DISCUSSION TITLE
what is the  best practices with mimir in production environment

### DISCUSSION THREAD

**Discussion by z7658329** [grafana/mimir#6602] [Help and support]
1. should we deploy mimir with microservices mode in production to support huge metric data？
2. Which deploy mode is recommended？deploy with Helm or  Puppet  or Jsonnet ?

first, the community is well documented，but I hope to provide an example of best practice,

**Comment by charleskorn** [comment]
>should we deploy mimir with microservices mode in production to support huge metric data？

What's your definition of "huge" here? Microservices mode is definitely the recommended option for large installations.

**Reply by z7658329** [reply]
means lots of，maybe  one billion even more；seems microservices mode works

**Reply by charleskorn** [reply]
At that scale, microservices mode is definitely recommended.

**Comment by charleskorn** [comment]
>Which deploy mode is recommended？deploy with Helm or Puppet or Jsonnet ?

Helm and Jsonnet are both supported by the Mimir maintainers. If you're already familiar with Helm, then that's likely the most straightforward option.

Puppet is support is maintained by the community.

**Reply by z7658329** [reply]
em, i dont familiar with Helm, I'd better look at the documentation first, thank you

**Comment by z7658329** [comment]
i have aked, helm/puppet/jsonnet  are not supported on internal deploy platforms！but we can build image by binary file, then we can deploy by inner platforms!

now i want to use microservices mode, how to deploy with mimir binary file in linux? refer to mimir architecture，there seems ten components：
Compactor、Distributor、Ingester、Querier、Query-frontend、Store-gateway、(Optional) Alertmanager、(Optional) Overrides-exporter
、(Optional) Query-scheduler、(Optional) Ruler

@charleskorn

**Reply by charleskorn** [reply]
This is not recommended - there are a number of good practices baked into the Helm chart and Jsonnet.

If you want to go down this path, I'd recommend looking at the manifests produced by the Helm chart or Jsonnet and use them as a base for running Mimir on bare metal.
